src/utils/refreshDockerode.ts

```diff
@@ -72,14 +75,31 @@ async function getDockerodeOptions(newEnv: NodeJS.ProcessEnv): Promise<DockerOpt
     return undefined;
 }
 
-async function getSshAuthSock(): Promise<string | undefined> {
-    if (isWindows()) {
-        return '\\\\.\\pipe\\openssh-ssh-agent';
-    } else {
+async function validateSshAuthSock(newEnv: NodeJS.ProcessEnv): Promise<boolean> {
+    if (!newEnv.SSH_AUTH_SOCK && isWindows()) {
+        // On Windows, we can use this one by default
+        newEnv.SSH_AUTH_SOCK = '\\\\.\\pipe\\openssh-ssh-agent';
+    } else if (!newEnv.SSH_AUTH_SOCK) {
         // On Mac and Linux, if SSH_AUTH_SOCK isn't set there's nothing we can do
         // Running ssh-agent would yield a new agent that doesn't have the needed keys
-        await ext.ui.showWarningMessage('In order to use an SSH DOCKER_HOST on OS X and Linux, you must configure an ssh-agent.');
+        return false;
     }
+
+    // Validate the pipe actually exists
+    return await new Promise<boolean>((resolve, reject) => {
+        const authPipe = new Socket();
+
+        authPipe.on('error', () => {
+            resolve(false);
+        });
+
+        authPipe.on('connect', () => {
+            authPipe.end();
+            resolve(true);
+        });
+
+        authPipe.connect(newEnv.SSH_AUTH_SOCK);

```

Comment:
So this takes forever, and `setTimeout` doesn't work--need to work out some other way.